### PR TITLE
overhaul of list property handling

### PR DESCRIPTION
**Pull request by mpollmeier**
* use IndexedSeq (backed by ArraySeq) everywhere
* use zero-copy ArraySeq.unsafeWrapArray / ArraySeq.unsafeArray as suggested by @bbrehm
* drop Cardinality.ISeq
* handle lists for edge properties as well
* provide general conversion function (depends on https://github.com/ShiftLeftSecurity/overflowdb/pull/206)
* generate domain starting point (equivalent of `Cpg.scala`)

### CODE REVIEW

File: codegen/src/main/scala/overflowdb/codegen/CodeGen.scala [hunk 1]
```diff
@@ -702,7 +772,7 @@ class CodeGen(schema: Schema) {
                    |      case _ => throw new MatchError("unreachable")
                    |    }}.toArray
                    |
-                   |  this._$memberName = if(arr == null) collection.immutable.ArraySeq.empty
+                   |  this._$memberName = if(arr == null) IndexedSeq.empty
```
Comment (bbrehm) [inline codegen/src/main/scala/overflowdb/codegen/CodeGen.scala:775]: Why this change? I'm not objecting, just asking. 

The original point of me using `ArraySeq.empty` instead of `IndexedSeq.empty` was to maintain an inofficial stronger API (we always return `ArraySeq`), even if we only promise `IndexedSeq` in the type signature.

Comment (mpollmeier) [reply codegen/src/main/scala/overflowdb/codegen/CodeGen.scala]: good idea, adjusted in db3d682

File: codegen/src/main/scala/overflowdb/codegen/CodeGen.scala [hunk 2]
```diff
@@ -196,16 +276,15 @@ class CodeGen(schema: Schema) {
...
-                 |}""".stripMargin
-            case Cardinality.ISeq =>
-              s"""def $nameCamelCase: $tpe = {
-                 |  val p = property("$name")
-                 |  if (p != null) p.asInstanceOf[java.util.List[${typeFor(property)}]].asScala.to(IndexedSeq)
-                 |  else IndexedSeq.empty
+                 |  property("$name") match {
+                 |    case null => IndexedSeq.empty
+                 |    case arr: Array[_] if arr.isEmpty => IndexedSeq.empty
+                 |    case arr: Array[Object] => scala.collection.immutable.ArraySeq.unsafeWrapArray(arr).asInstanceOf[$returnType]
```
[the remark this answers is not in the document]
Comment (mpollmeier) [reply codegen/src/main/scala/overflowdb/codegen/CodeGen.scala:284]: good point, yes indeed a primitive array wouldn't match here, I'll change it to `Array[_]` and check why this wasn't covered by tests

[the remark this answers is not in the document]
Comment (mpollmeier) [reply codegen/src/main/scala/overflowdb/codegen/CodeGen.scala]: fixed in 2b56abfd75b1406e3386a52fd409f741bcdfab57

**Review by bbrehm — APPROVED** [review]
